src/main/java/org/joda/beans/PropertyPath.java

```diff
@@ -0,0 +1,269 @@
+            if (entryStr.endsWith("]") && start > 0) {
+                key = entryStr.substring(start + 1, entryStr.length() - 1);
+                if (key.length() == 0) {
+                    throw new IllegalArgumentException("Invalid property path, empty key: " + propertyPath);
+                }
+                char firstChar = key.charAt(0);
+                index = -1;
+                if (firstChar == '-' || (firstChar >= '0' && firstChar <= '9')) {
+                    try {
+                        index = Integer.parseInt(key);
+                    } catch (NumberFormatException ex) {
+                        // index = -1
+                    }
+                }
+                propName = entryStr.substring(0, start);
+            }
+            return new PathEntry(propName, key, index);
+        }
+
+        private PathEntry(String propertyName, String key, int index) {
+            this.propertyName = propertyName;
+            this.key = key;
+            this.index = index;
+        }
+
+        private Object get(Bean bean) {
+            try {
+                return bean.metaBean().metaProperty(propertyName).get(bean);
+            } catch (RuntimeException ex) {
+                return null;
+            }
+        }
+
+        private Object extract(Object obj) {
+            // maps can be queried using the [key] suffix if desired
+            // an [index] suffix will be queried as a key, not an index
+            if (obj instanceof Map<?, ?>) {
+                if (key == null) {
+                    return extract(((Map<?, ?>) obj).values());
+                } else {
+                    Map<?, ?> map = ((Map<?, ?>) obj);
+                    for (Entry<?, ?> mapEntry : map.entrySet()) {
+                        if (key.equals(mapEntry.getKey())) {
+                            return mapEntry.getValue();
+                        }
+                    }
+                    return null;
+                }
+            }
+
+            // lists/sets can be queried using the [index] suffix if desired
+            if (obj instanceof Iterable<?>) {
+                if (key != null && index < 0) {
+                    return null;
+                }
+                if (obj instanceof List<?>) {
+                    List<?> list = (List<?>) obj;
+                    if (index < list.size()) {
+                        return list.get(index);
+                    }
+                    return null;
+                }
+                Iterator<?> it = ((Iterable<?>) obj).iterator();
+                int i = 0;
+                while (it.hasNext() && i < index) {
+                    it.next();
+                    i++;
+                }
+                return it.hasNext() ? it.next() : null;
+            }
+
+            // not a collection
+            if (key != null && !"0".equals(key)) {
+                return null;
+            }
+            return obj;
+        }
+    }
+
+}
```

Comment:
This suggestion is wrong, because it would change the behaviour of the code. The code is specifically designed to not throw an exception at line 201. Can you approve the PR for me?

Reply:
Please approve the PR